tests/tests_ssh.yml

```diff
@@ -30,6 +30,20 @@
       hostvars[kdump_test_ssh_server_outside]['ansible_default_ipv4']['address']
       }}
 
+    # This is the outside address. Ansible will connect to it to
+    # copy the ssh key.
+    kdump_ssh_server: "{{ kdump_test_ssh_server_outside }}"
+    kdump_ssh_user: >-
+      {{ hostvars[kdump_test_ssh_server_outside]['ansible_user_id'] }}
+    kdump_path: /tmp/test
+    kdump_target:
+      type: ssh
+      # This is the ssh dump server address visible from inside
+      # the machine being configured. Dumps are to be copied
+      # there.
+      location: "{{ kdump_ssh_user }}@{{ kdump_test_ssh_server_inside }}"
+    kdump_reboot_ok: true
```

Comment:
I think that you need to set some tag to indicate that the role reboots the managed host, otherwise it will not work when executed against localhost (when managed host == control host), because you will kill the ansible-playbook process itself (as can happen in some CI systems that do not provide a separate Ansible controller).

Reply:
The CI system uses `tests::reboot` to tag tests that will cause the node to reboot

Reply:
yes, and if one adds the tag to `tasks/execute_with_reboot.yml`, it will be visible in any playbook that imports (not includes) this task file.

Reply:
These tests require reboot on systems that did not have kdump enabled prior to running the test. Because kdump was not enabled and never started, it never reserved memory for  the crash kernel.
This happens on RHEL 6 image, even though as @richm  tested in https://github.com/linux-system-roles/kdump/pull/76#issuecomment-801168768
To sum up, the requirement to reboot is not consistent and depends rather on the way the image is built than the OS itself. Therefore, instead of adding a tag, I am adding a conditional `meta: end_host` as @pcahyna suggested.